tasks/main.yml

```diff
@@ -3,5 +3,13 @@
 - name: Set ansible_facts required by role
   include_tasks: set_vars.yml
 
+- name: Handle insights unregistration
+  include_tasks: insights-client-unregistration.yml
+  when: rhc_insights.state | d('present') == "absent"
```

Comment:
- I think we need to limit this also to RHEL, ie `ansible_distribution == "RedHat"`, as `insights-client` exists only there
- in addition to check for `rhc_insights.state`, I think we need to unregister from Insights also when `rhc_state | d("present") == "absent` -- it does not make sense to not unregister from Insights when unregistering from rhsm
   - this will also allow us to drop the manually specified `rhc_insights.state=absent` when `rhc_state=absent` in the tests
- minor nitpick: double quotes instead of single ones